automation/render-local-env.sh

```diff
@@ -1,17 +1,82 @@
 #!/usr/bin/env bash
+set -e
 
-WORKING_DIR=${1-$(pwd)}
-VAULT_TOKEN=$2
-ENV=$3
-LOCAL_UI=false
-LOCAL_UI_PARAM=$4
-if [ "$LOCAL_UI_PARAM" = "local_ui" ]; then
-  LOCAL_UI=true 
-fi
-
-# render ctmpls
-docker pull broadinstitute/dsde-toolbox:dev
-docker run --rm -e VAULT_TOKEN=${VAULT_TOKEN} \
-    -e ENVIRONMENT=${ENV} -e ROOT_DIR=${WORKING_DIR} -v ${WORKING_DIR}:/working \
-    -e OUT_PATH=/working/src/test/resources -e INPUT_PATH=/working/docker -e LOCAL_UI=$LOCAL_UI \
-    broadinstitute/dsde-toolbox:dev render-templates.sh
+## Run from automation/
+## Clones the firecloud-automated-testing repo, pulls templatized configs, and renders them to src/test/resources
+
+# Defaults
+WORKING_DIR=$PWD
+VAULT_TOKEN=$(cat ~/.vault-token)
+FIRECLOUD_AUTOMATED_TESTING_BRANCH=master
+ENV=dev
+SERVICE=leonardo
+LOCAL_UI=${LOCAL_UI:-false}  # local ui defaults to false unless set in the env
+
+# Parameters
+FIRECLOUD_AUTOMATED_TESTING_BRANCH=${1:-$FIRECLOUD_AUTOMATED_TESTING_BRANCH}
+WORKING_DIR=${2:-$WORKING_DIR}
+VAULT_TOKEN=${3:-$VAULT_TOKEN}
+ENV=${4:-$ENV}
+
+confirm () {
+    # call with a prompt string or use a default
+    read -r -p "${1:-Are you sure?} [y/N] " response
+    case $response in
+        [yY])
+            shift
+            $@
+            ;;
+        *)
+            ;;
+    esac
+}
+
+# clone the firecloud-automated-testing repo
+clone_repo() {
+    original_dir=$PWD
+    if [[ $PWD == *"${SERVICE}"* ]]
+        then
+        cd ..
+    fi
+
+    echo "Currently in" $PWD
+    confirm "OK to clone here?" git clone https://github.com/broadinstitute/firecloud-automated-testing.git
+
+    cd $original_dir
+}
+
+pull_configs() {
+    original_dir=$WORKING_DIR
+    if [[ $PWD == *"${SERVICE}"* ]]; then
+        cd ../..
+    fi
+    cd firecloud-automated-testing
+    echo $PWD
+    git pull
+    git stash
+    git checkout ${FIRECLOUD_AUTOMATED_TESTING_BRANCH}
```

Comment:
Where does this checkout to, and how is it used by the leo repo?

Reply:
it checks out to the `firecloud-automated-testing` directory (cloned at the root if it doesn't exist in the `clone_repo` step, and then leo uses the config ctmpls in that repo to render configs.  the main goal of the ticket this was a part of was to put all those testing configs in one place (bc they are common) and pull from that place, rather than have copies scattered among the repos

Reply:
But are you expecting `firecloud-automated-testing` to be in a subdirectory of `leonardo/automation`?  How can that work, since we're already in a git repo?

Or should it be in some other location?  How does the script find it in that case?

Reply:
nope i'm expecting it to exist in the directory above `leonardo`.  this script is expected to be run from `leonardo/automation`, and the `clone_repo` function takes care of  putting `firecloud-automated-testing` in the right place if it doesn't already exist